### PR TITLE
Layer: render new layers in a host container div.

### PR DESCRIPTION
**Pull request by dzearing**
Layers which are appended to the end of the document end up invalidating the `body` element, which causes massive recomputation of layout.

Now with the layer appends scoped within a div, the browser can isolate recomputation to only the container div, resulting in faster ContextualMenu rendering.

### CODE REVIEW

File: packages/react/src/components/Layer/Layer.base.tsx [hunk 1]
```diff
@@ -54,7 +54,19 @@ export const LayerBase: React.FunctionComponent<ILayerProps> = React.forwardRef<
         return doc?.getElementById(hostId) ?? null;
       } else {
         const defaultHostSelector = getDefaultTarget();
-        return (defaultHostSelector ? doc?.querySelector(defaultHostSelector) : doc?.body) ?? null;
+
+        let host = defaultHostSelector && doc?.querySelector(defaultHostSelector);
```
Comment (ThomasMichon) [inline packages/react/src/components/Layer/Layer.base.tsx:58]: I suggest defining `exports` for `createDefaultLayerHost` and `cleanupDefaultLayerHost`.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/855ec38026ff54ddf929a9e540abbf58d5b6ba78'>855ec38026ff54ddf929a9e540abbf58d5b6ba78</a></sub>
<div id="bundleSizeComment9423"/>

**Review by RobMeyer — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/BaseButton-mount_base.html">776</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/BaseButton-mount.html">776</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Breadcrumb-mount_base.html">2356</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Breadcrumb-mount.html">2367</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Checkbox-mount_base.html">1288</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Checkbox-mount.html">1286</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/CheckboxBase-mount_base.html">1063</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/CheckboxBase-mount.html">1073</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ChoiceGroup-mount_base.html">4067</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ChoiceGroup-mount.html">4028</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ComboBox-mount_base.html">855</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ComboBox-mount.html">862</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/CommandBar-mount_base.html">9046</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/CommandBar-mount.html">9080</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ContextualMenu-mount_base.html">10252</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ContextualMenu-mount.html">10230</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DefaultButton-mount_base.html">986</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DefaultButton-mount.html">972</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DetailsRow-mount_base.html">3253</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DetailsRow-mount.html">3322</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DetailsRowFast-mount_base.html">3267</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DetailsRowFast-mount.html">3305</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DetailsRowNoStyles-mount_base.html">3086</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DetailsRowNoStyles-mount.html">3135</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Dialog-mount_base.html">1898</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Dialog-mount.html">1922</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DocumentCardTitle-mount_base.html">136</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/DocumentCardTitle-mount.html">148</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Dropdown-mount_base.html">2828</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Dropdown-mount.html">2813</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/FocusTrapZone-mount_base.html">1561</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/FocusTrapZone-mount.html">1567</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/FocusZone-mount_base.html">1570</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/FocusZone-mount.html">1582</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/IconButton-mount_base.html">1529</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/IconButton-mount.html">1500</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Label-mount_base.html">294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Label-mount.html">303</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Layer-mount_base.html">2525</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Layer-mount.html">2548</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Link-mount_base.html">406</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Link-mount.html">394</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/MenuButton-mount_base.html">1256</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/MenuButton-mount.html">1281</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/MessageBar-mount_base.html">1803</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/MessageBar-mount.html">1812</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Nav-mount_base.html">2851</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Nav-mount.html">2845</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/OverflowSet-mount_base.html">916</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/OverflowSet-mount.html">954</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Panel-mount_base.html">1807</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Panel-mount.html">1812</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Persona-mount_base.html">837</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Persona-mount.html">854</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Pivot-mount_base.html">1260</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Pivot-mount.html">1246</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/PrimaryButton-mount_base.html">1093</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/PrimaryButton-mount.html">1113</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Rating-mount_base.html">6619</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Rating-mount.html">6682</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/SearchBox-mount_base.html">1107</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/SearchBox-mount.html">1103</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Shimmer-mount_base.html">2188</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Shimmer-mount.html">2146</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Slider-mount_base.html">1681</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Slider-mount.html">1666</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/SpinButton-mount_base.html">4357</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/SpinButton-mount.html">4361</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Spinner-mount_base.html">383</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Spinner-mount.html">388</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/SplitButton-mount_base.html">2736</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/SplitButton-mount.html">2739</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Stack-mount_base.html">445</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Stack-mount.html">463</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/StackWithIntrinsicChildren-mount_base.html">1973</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/StackWithIntrinsicChildren-mount.html">1962</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/StackWithTextChildren-mount_base.html">4513</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/StackWithTextChildren-mount.html">4509</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/SwatchColorPicker-mount_base.html">10025</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/SwatchColorPicker-mount.html">9953</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/TagPicker-mount_base.html">2319</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/TagPicker-mount.html">2336</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/TeachingBubble-mount_base.html">85734</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/TeachingBubble-mount.html">83640</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Text-mount_base.html">361</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Text-mount.html">367</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/TextField-mount_base.html">1210</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/TextField-mount.html">1191</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ThemeProvider-mount_base.html">1009</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ThemeProvider-mount.html">1020</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ThemeProvider-virtual-rerender_base.html">558</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ThemeProvider-virtual-rerender.html">570</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1574</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1605</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Toggle-mount_base.html">689</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/Toggle-mount.html">686</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/buttonNative-mount_base.html">109</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22295/perf-test/buttonNative-mount.html">113</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/Layer/Layer.base.tsx (hunk 1 above)
Comment (dzearing) [reply packages/react/src/components/Layer/Layer.base.tsx:58]: Updating incoming.

Comment (dzearing) [reply packages/react/src/components/Layer/Layer.base.tsx]: done.

**Review by RobMeyer — APPROVED** [review]

File: packages/react/src/components/Layer/Layer.notification.ts [hunk 2]
```diff
@@ -112,6 +113,32 @@ export function unregisterLayerHost(hostId: string, layerHost: ILayerHost): void
   }
 }
 
+/**
+ * When no default layer host is provided, this function is executed to create the default host.
+ */
+export function createDefaultLayerHost(doc: Document): Node | null {
+  const host = doc.createElement('div');
+  host.setAttribute('id', 'fluent-default-layer-host');
```
Comment (spmonahan) [inline packages/react/src/components/Layer/Layer.notification.ts:121]: nit: Replace `'fluent-default-layer-host'` with `defaultHostId`.

**Review by spmonahan — APPROVED** [review]

**Review by RobMeyer — APPROVED** [review]

File: packages/react/src/components/Layer/Layer.notification.ts (hunk 2 above)
Comment (dzearing) [reply packages/react/src/components/Layer/Layer.notification.ts:121]: Oh oops! meant to do that!

Comment (dzearing) [reply packages/react/src/components/Layer/Layer.notification.ts]: done.

**Review by RobMeyer — APPROVED** [review]

**Review by ThomasMichon — APPROVED** [review]

**Comment by hecerinc** [comment]
@dzearing I think this PR broke modeless Dialogs as well (see [here](https://fabricweb.z5.web.core.windows.net/oufr/8.64.0/#/examples/dialog))

**Comment by jacksonv1lle** [comment]
This change has caused a few problems with our SharePoint libraries since some of our 3rd party libraries (and SharePoint itself) are still using an old version of Fluent-UI. The layers are appearing behind one another. 

Would it be possible to implement this feature using a new property in the ILayerProps so the developer can opt-in? That way we would have backwards compatibility when there are multiple versions of Fluent-UI being used.

**Pull request merged**